### PR TITLE
Adding SendGrid Extension

### PR DESCRIPTION
**Pull request by soninaren**

### CODE REVIEW

File: DotNetWorker.sln [hunk 1]
```diff
@@ -1,264 +1,271 @@
...
+		{248ACC90-D4F9-432B-B7B9-341EF00FA072} = {083592CA-7DAB-44CE-8979-44FAFA46AEC3}
+		{23AAA8FA-5B51-49BF-8021-148C8031A321} = {083592CA-7DAB-44CE-8979-44FAFA46AEC3}
+		{BC56E506-A2F7-46D4-95DC-BD97E5AF92D4} = {9D6603BD-7EA2-4D11-A69C-0D9E01317FD6}
+		{2A72383A-9C00-41FB-9D29-909EE5E6BFB9} = {9D6603BD-7EA2-4D11-A69C-0D9E01317FD6}
+		{C4C0CE7E-6A34-473E-A8FB-7D8FBA765779} = {9D6603BD-7EA2-4D11-A69C-0D9E01317FD6}
+		{035AAD56-5F51-476C-8556-0F6CFD6EF1BF} = {9D6603BD-7EA2-4D11-A69C-0D9E01317FD6}
+		{49CB0588-5243-4616-9A21-C52CA90599BC} = {A7B4FF1E-3DF7-4F28-9333-D0961CDDF702}
+	EndGlobalSection
+	GlobalSection(ExtensibilityGlobals) = postSolution
+		SolutionGuid = {497D2ED4-A13E-4BCA-8D29-F30CA7D0EA4A}
```
Comment (fabiocav) [inline DotNetWorker.sln:269]: Something is off with the solution change here

File: extensions/Worker.Extensions.SendGrid/SendGridAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.Functions.Worker.Extensions.Abstractions;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// Attribute used to bind a parameter to a SendGridMessage that will automatically be
```
Comment (fabiocav) [inline extensions/Worker.Extensions.SendGrid/SendGridAttribute.cs:10]: This description doesn't apply as we won't support the binding to `SendGridMessage`

File: extensions/Worker.Extensions.SendGrid/Worker.Extensions.SendGrid.csproj [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <AssemblyName>Microsoft.Azure.Functions.Worker.Extensions.SendGrid</AssemblyName>
+    <RootNamespace>Microsoft.Azure.Functions.Worker.Extensions.SendGrid</RootNamespace>
+    <Description>Azure Service Bus extensions for .NET isolated functions</Description>
```
Comment (fabiocav) [inline extensions/Worker.Extensions.SendGrid/Worker.Extensions.SendGrid.csproj:6]: Update description to sendgrid

File: extensions/Worker.Extensions.SendGrid/Worker.Extensions.SendGrid.csproj [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  <PropertyGroup>
+    <AssemblyName>Microsoft.Azure.Functions.Worker.Extensions.SendGrid</AssemblyName>
+    <RootNamespace>Microsoft.Azure.Functions.Worker.Extensions.SendGrid</RootNamespace>
+    <Description>Azure Service Bus extensions for .NET isolated functions</Description>
+
+    <!--Version information-->
+    <VersionPrefix>3.0.2</VersionPrefix>
+
+    <!--Temporarily opting out of documentation. Pending documentation-->
+    <GenerateDocumentationFile>false</GenerateDocumentationFile>
```
Comment (fabiocav) [inline extensions/Worker.Extensions.SendGrid/Worker.Extensions.SendGrid.csproj:12]: If all the docs are done, remove this.

Comment (soninaren) [reply extensions/Worker.Extensions.SendGrid/Worker.Extensions.SendGrid.csproj:12]: Copied the flag from other extension. Not sure which doc is being referenced here.

Comment (fabiocav) [reply extensions/Worker.Extensions.SendGrid/Worker.Extensions.SendGrid.csproj:12]: This is about the XML docs applied to the code. Looks like the attribute is documented, so this should be removed.

**Review by fabiocav — APPROVED** [review]
Just one minor follow up on the doc question. Flagging as approved, but it would be good to change that.

**Pull request merged**